src/main.js

```diff
@@ -6,59 +6,57 @@ var ideaText = document.querySelector('.card-idea-text');
 var bubbleParent = document.querySelector('.purple-1');
 var ideaForm = document.querySelector('.idea-form');
 var allCards = document.querySelector('.all-cards');
+var deleteIcon = document.querySelector('.card-delete-icon')
 
 var storage = [];
 
 bubbleParent.addEventListener('click', clickHandler);
-ideaForm.addEventListener('keypress', keyHandler);
-window.addEventListener('load', keyHandler);
-//saveButtonDisable();
+window.addEventListener('keypress', keyHandler);
 
 function clickHandler(event) {
-  if (event.target.className === "save-button") saveIdeaButton()
+  if (event.target.className === "save-button") saveIdeaButton();
+  if (event.target.className === "card-delete-icon") deleteIdea();
 }
 
 function keyHandler(event) {
-  if (event.target.className === "title-input") saveButtonDisable()
-  if (event.target.className === "body-input") saveButtonDisable()
+  if (event.target.className === "title-input") saveButtonDisable();
+  if (event.target.className === "body-input") saveButtonDisable();
 }
 
 function saveIdeaButton() {
-  var obj = newIdeaInstantiation(titleInput.value, bodyInput.value)
-  obj.saveToStorage()
-  displayCard(buildCard(obj))
+  var newIdea = new Idea (titleInput.value, bodyInput.value)
+  newIdea.saveToStorage()
+  buildCard(newIdea)
   resetForm()
-}
-
-function newIdeaInstantiation(title, body) {
-  var newIdea = new Idea (title, body)
-  return newIdea
+  pushToStorage(newIdea)
 }
 
 
 function buildCard(obj) {
-  return ` <section class="card" id="${obj.id}">
-    <section class="card-header">
-      <img src="./img/star.svg" class="card-star-favorite-icon" alt="star favorite">
-      <img src="./img/delete.svg" class="card-delete-icon" alt="delete button">
-    </section>
-    <section class="card-body">
-      <h3 class="card-idea-title">
-        ${obj.title}
-      </h3>
-      <p class="card-idea-text">
-        ${obj.body}
-      </p>
-    </section>
-    <section class="card-footer">
-      <img src="./img/comment.svg" class="card-add-comment-icon" alt="add comment button">
-      <p>Comment</p>
+  allCards.insertAdjacentHTML('afterbegin',`
+    <section class="card" data.id="${obj.id}">
+      <section class="card-header">
+        <img src="./img/star.svg" class="card-star-favorite-icon" alt="star favorite">
+        <img src="./img/delete.svg" class="card-delete-icon" alt="delete button">
+      </section>
+      <section class="card-body">
+        <h3 class="card-idea-title">
+          ${obj.title}
+        </h3>
+        <p class="card-idea-text">
+          ${obj.body}
+        </p>
+      </section>
+      <section class="card-footer">
+        <img src="./img/comment.svg" class="card-add-comment-icon" alt="add comment button">
+        <p>Comment</p>
+      </section>
```

Comment:
This ended up being quite the strange bug, didn't it? It had us pretty baffled for a long time. Still not even 100% sure why this fixed it.